playbooks/cnf/deploy-run-cnf-tests-script.yaml

```diff
@@ -0,0 +1,193 @@
+          register: cnftests_git_clone
+          retries: 3
+          delay: 5
+          until: cnftests_git_clone is not failed
+
+      rescue:
+        - name: Git clone cnf-feature-deploy repo master branch
+          ansible.builtin.git:
+            repo: "{{ cnf_test_repo }}"
+            dest: "{{ cnf_test_dir }}/{{ cnftests_git_dest }}"
+            force: true
+            update: true
+            depth: 1
+            recursive: true
+            version: "master"
+          register: cnftests_master_git_clone
+          retries: 3
+          delay: 5
+          until: cnftests_master_git_clone is not failed
+
+    - name: Install Ginkgo v2
+      changed_when: false
+      ansible.builtin.shell: |
+        go install {{ ginkgo_link }}
+      environment:
+        PATH: "/usr/local/go/bin:{{ ansible_env.PATH }}"
+
+    - name: Get branch information from MetalLB project
+      ansible.builtin.uri:
+        url: "https://github.com/openshift/metallb-operator/tree/release-{{ version }}"
+        method: GET
+      register: metallb_github_response
+
+    - name: Set MetalLB branch variable based on existence
+      ansible.builtin.set_fact:
+        dynamic_metallb_branch: "{{ 'release-' + version if metallb_github_response.status == 200 else 'main' }}"
+
+    - name: Get branch information from SR-IOV project
+      ansible.builtin.uri:
+        url: "https://github.com/openshift/sriov-network-operator/tree/release-{{ version }}"
+        method: GET
+      register: sriov_github_response
+
+    - name: Set SR-IOV branch variable based on existence
+      ansible.builtin.set_fact:
+        dynamic_sriov_branch: "{{ 'release-' + version if sriov_github_response.status == 200 else 'master' }}"
+
+    - name: Get branch information from Node Tuning operator project
+      ansible.builtin.uri:
+        url: "https://github.com/openshift/cluster-node-tuning-operator/tree/release-{{ version }}"
+        method: GET
+      register: tuning_github_response
+
+    - name: Set Node Tuning operator branch variable based on existence
+      ansible.builtin.set_fact:
+        dynamic_tuning_branch: "{{ 'release-' + version if tuning_github_response.status == 200 else 'master' }}"
+
+    - name: Update all submodules
+      changed_when: false
+      environment:
+        METALLB_OPERATOR_TARGET_COMMIT: "{{ dynamic_metallb_branch }}"
+        SRIOV_NETWORK_OPERATOR_TARGET_COMMIT: "{{ dynamic_sriov_branch }}"
+        CLUSTER_NODE_TUNING_OPERATOR_TARGET_COMMIT: "{{ dynamic_tuning_branch }}"
+        PATH: "/usr/local/go/bin:{{ ansible_env.PATH }}"
+      ansible.builtin.shell: |
+        make init-git-submodules
+      args:
+        chdir: "{{ cnf_test_dir }}/{{ cnftests_git_dest }}"
+
+    - name: List cnf test image versions
+      changed_when: false
+      ansible.builtin.shell:
+        cmd: |
+          set -o pipefail
+          brew list-builds --package=cnf-tests-container -r --state=COMPLETE | grep "v{{ version }}"
```

Comment:
Do we have `brew` installed in the bastion?

Reply:
Yes
 https://github.com/openshift-kni/eco-ci-cd/blob/main/playbooks/infra/deploy-vm-bastion-libvirt.yml#L346